### PR TITLE
FIX:  FIX: bolt11 links are opening the app but not the invoice itsel…

### PR DESCRIPTION
**Pull request by marcosrdz**
…f #1352 #1359

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/rt40ttxqd7k2xrp31nnuqmdk50

[download apk](http://filestorage.bluewallet.io:1488/1595637496.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/abgy719rk0v0uve2nqppqtrkn0

[download apk](http://filestorage.bluewallet.io:1488/1595646386.apk)

**Comment by Overtorment** [comment]
I believe the problem was that BW was not recognizing URLs with `//`, but other URLs work fine even when the app is running:

bolt 11 Peek 2020-07-25 17-32


and those URLs are not used in real life.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/j2t9k2b639ta9ny9jg9vhdbdnc

[download apk](http://filestorage.bluewallet.io:1488/1595723813.apk)

File: App.js [hunk 1]
```diff
@@ -30,6 +30,52 @@ const lightningModalString = 'Lightning Invoice';
...
 const notifications = require('./blue_modules/notifications'); // eslint-disable-line no-unused-vars
+const linking = {
+  prefixes: ['bitcoin:', 'bitcoin://', 'lightning:', 'lightning://'],
+  config: {
+    screens: {
+      SendDetailsRoot: {},
+    },
+  },
+  getStateFromPath: (path, options) => {
+    path = path.replace('//', '');
```
Comment (Overtorment) [inline App.js:41]: should be hidden inside deeplinking class and unit tested

File: class/deeplink-schema-match.js [hunk 2]
```diff
@@ -76,25 +76,33 @@ class DeeplinkSchemaMatch {
         },
       ]);
     } else if (DeeplinkSchemaMatch.isBitcoinAddress(event.url)) {
-      completionHandler([
-        'SendDetailsRoot',
-        {
-          screen: 'SendDetails',
-          params: {
-            uri: event.url,
+      if (typeof navigator !== 'undefined' && navigator.product === 'ReactNative') {
```
Comment (Overtorment) [inline class/deeplink-schema-match.js:79]: very bad

File: screen/lnd/lndViewInvoice.js [hunk 3]
```diff
@@ -323,7 +323,7 @@ export default class LNDViewInvoice extends Component {
 
             <SecondButton
               onPress={() => {
-                Share.open({ message: `lightning:${invoice.payment_request}` }).catch(error => console.log(error));
+                Share.open({ message: `lightning://${invoice.payment_request}` }).catch(error => console.log(error));
```
Comment (Overtorment) [inline screen/lnd/lndViewInvoice.js:326]: still not sure https://github.com/bitcoin/bips/blob/master/bip-0021.mediawiki

File: screen/receive/details.js [hunk 4]
```diff
@@ -315,7 +315,8 @@ const ReceiveDetails = () => {
   };
 
   const handleShareButtonPressed = () => {
-    Share.open({ message: bip21encoded }).catch(error => console.log(error));
+    const message = bip21encoded.replace('bitcoin:', 'bitcoin://');
```
Comment (Overtorment) [inline screen/receive/details.js:318]: still not sure https://github.com/bitcoin/bips/blob/master/bip-0021.mediawiki

**Comment by Overtorment** [comment]
@limpbrains what are your thoughts on this?

**Comment by marcosrdz** [comment]
> I agree with @Overtorment that checking for globals inside DeeplinkSchemaMatch 
> 
> `if (typeof navigator !== 'undefined' && navigator.product === 'ReactNative') {`
> 
> looks hackish, should be changed and well tested. 
> 
> 
> 
> If it is impossible to archive what's needed with `DeeplinkSchemaMatch`, let's migrate to `<NavigationContainer linking={linking}` for all links to have a tested, consisted approach. Is it possible @marcosrdz ?
> 
> 
> 
> @Overtorment proved that there are no use of links with double slashes like `lightning://` or `bitcoin://`. I've found another one [BOLT11](https://github.com/lightningnetwork/lightning-rfc/blob/master/11-payment-encoding.md), look for **use 'lightning:' as a prefix before the BOLT-11 encoding (note: not 'lightning://')**. Therefore, we don't need to worry about links with `//` and this code should be deleted to reduce complexity.

The goal is to be able to handle deeplink calls from apps like notes, mail, messages, messenger, whatsapp and others. Right now the app opens but doesn't handle them.

**Comment by Overtorment** [comment]
cleanup

**Pull request closed without merging**